v1/plugin/plugin.go

```diff
@@ -328,5 +384,186 @@ func startPlugin(srv server, m *meta, p *pluginProxy) int {
 	go p.HeartbeatWatch()
 	// TODO(danielscottt): exit code
 	<-p.halt
-	return 0
+
+	return nil
+}
+
+// GetPluginType converts a pluginType to a string
+// describing what type of plugin this is
+func getPluginType(plType pluginType) string {
+	switch plType {
+	case collectorType:
+		return "collector"
+
+	case publisherType:
+		return "publisher"
+
+	case processorType:
+		return "processor"
+	}
+	return ""
+}
+
+// GetPluginType converts a pluginType to a string
+// as described in snap/control/plugin/plugin.go
+func getRPCType(i int) string {
+	if i == 0 {
+		return "NativeRPC (0)"
+	} else if i == 2 {
+		return "GRPC (2)"
+	}
+	return "Not a valid RPC Type"
+}
+
+func showDiagnostics(m meta, p *pluginProxy, c Config) error {
+	defer timeTrack(time.Now(), "showDiagnostics")
+	printRuntimeDetails(m)
+	err := printConfigPolicy(p, c)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
This is particularly for the case when printConfigPolicy errors from a missing required config. It will show warning there then in printMetricTypes an error will be thrown from something not being able to load properly without the required config. This is intentional.